### PR TITLE
Run tests when ci:run label is added to a pull request.

### PR DESCRIPTION
**Pull request by jwithers**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,18 +3,32 @@ on:
...
-      - name: Show Workspace Context
-        run: echo 'Workspace is ${{ github.workspace }}'
-      - name: Nasty Hack To Get Right Docker Build Context
-        run: cp ./.github/actions/docker_action/Dockerfile . && cp ./.github/actions/docker_action/action.yml .
-      - name: Run Docker Tests
-        uses: ./
-      - name: Show commit
-        run: git show -q
+      - name: test_all.sh running from Dockerfile.micro
+        uses: docker://jpwithers/tflite-micro-tests
```
Comment (advaitjain) [inline .github/workflows/main.yml:14]: I'm assuming that we will be moving this to a docker image hosted with GitHub packages in a follow-up change?

I have created https://github.com/tensorflow/tflite-micro/issues/11 to start having open issues for work that we are punting for later. It would be nice to add a TODO(#11) comment in here as well.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -3,18 +3,32 @@ on:
...
+#       uncomment below to run only the five minute test bit.
+#        with:
+#          args: /bin/sh -c "/opt/tflm/tensorflow/lite/micro/tools/ci_build/test_all.sh"
```
Comment (advaitjain) [inline .github/workflows/main.yml:17]: is comment still valid? I would expect that directly running test_all.sh would be the same as using Dockerfile.micro.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -3,18 +3,32 @@ on:
...
+      - name: add a file for testing
+        run: git show -q > .commitinfo
+      - name: Commiting CI Variables Files
+        uses: EndBug/add-and-commit@3276c159cee23546e7fef9c8f9997e923587b543
```
Comment (advaitjain) [inline .github/workflows/main.yml:21]: this is ok to use but let's start adding some notes on what 3rd party github actions we are using (in addition to the non-trivial things that the CI system is doing). I think adding a top-level continuous_integration.md and putting notes in there would be very useful. We can clean it up later but documenting as we go along would be really nice.

File: main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+on:
```
Comment (advaitjain) [inline main.yml:1]: is this file needed? looks like a copy of .github/workflows/main.yml

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (jwithers) [reply .github/workflows/main.yml:17]: It is valid. Those lines override the CMD in the container which removes the GITHUB_PRESUBMIT argument, thus cutting out the bazel stuff.

File: .github/workflows/main.yml (hunk 1 above)
Comment (jwithers) [reply .github/workflows/main.yml:14]: Yeah, I am still sorting through that now and getting it set up on my fork.

File: main.yml (hunk 4 above)
Comment (jwithers) [reply main.yml:1]: Yeah, that's an error. Thanks.

File: .github/workflows/main.yml (hunk 2 above)
Comment (advaitjain) [reply .github/workflows/main.yml:17]: ah, yes.

I'm going to be soon enabling many more tests from test_all.sh so its time will also increase.

If you need something quick for your work, I might suggest changing this to be  /opt/tflm/tensorflow/lite/micro/tools/ci_build/test_x86.sh (with a comment that says, uncomment to run only the x86 tests for faster development of the CI work)

**Review by advaitjain — DISMISSED** [review]

**Review by advaitjain — APPROVED** [review]
